xenserver/tests/test_tasks.py

```diff
@@ -3,186 +3,265 @@
 """
 
 import pytest
+from testtools.assertions import assert_that
```

Comment:
I prefer to use `assert_that` (and I actually asked for it when I first looked at testtools for various things) because historically we've often needed to use someone else's base test classes (usually Django or Twisted).

Where possible, I prefer not to use base `TestCase` classes at all because they encourage putting utility functions (helpers, assertions, etc.) on the class, which leads to multiple almost-but-not-quite-the-same implementations across tests.